src/csharp/Microsoft.Spark/Utils/AssemblyLoader.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Spark.Utils
+{
+    internal static class AssemblyLoader
+    {
+        internal static Func<string, Assembly> LoadFromFile { get; set; } = Assembly.LoadFrom;
+
+        internal static Func<string, Assembly> LoadFromName { get; set; } = Assembly.Load;
+
+        private static readonly Dictionary<string, Assembly> s_assemblyCache =
+            new Dictionary<string, Assembly>();
+
+        private static readonly string[] s_searchPaths =
+            new[] { Directory.GetCurrentDirectory(), AppDomain.CurrentDomain.BaseDirectory };
+
+        private static readonly string[] s_extensions =
+            RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ?
+                    new[] { ".dll", ".exe", ".ni.dll", ".ni.exe" } :
+                    new[] { ".dll", ".ni.dll" };
+
+        private static ResolveEventHandler s_eventHandler = null;
+
+        private static readonly object s_cacheLock = new object();
+
+        internal static void InstallHandler()
+        {
+            if (s_eventHandler == null)
+            {
+                s_eventHandler = new ResolveEventHandler(ResolveAssembly);
+                AppDomain.CurrentDomain.AssemblyResolve += s_eventHandler;
+            }
+        }
+
+        internal static void RemoveHandler()
```

Comment:
Who calls this method?

Reply:
No one calls at the moment.  I'll remove.

Reply:
I think it might be better to be symmetrical between .NET Core and .NET Framework and have the same class responsible for hooking up the AssemblyResolve event.

---
In reply to: [306890361](https://github.com/dotnet/spark/pull/180#discussion_r306890361) [](ancestors = 306890361)

Reply:
Sounds good, I added the AssemblyResolve event hookup code all in one place.